lib/msal-core/FAQ.md

```diff
@@ -0,0 +1,224 @@
+
+The tokens returned by Azure AD have a default lifetime of 1 hour. However, as long as the user is active on your application and a valid Azure AD session exists in the browser, the acquireTokenSilent method can be used to renew tokens. The Azure AD session is valid for 24 hours and can be extended by the user by choosing "Keep me signed in" at sign-in time. For more details, read the [token and session lifetimes](https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-configurable-token-lifetimes) document.
+
+## How to get single sign-on in my application with MSAL.js
+
+Please read [Single Sign-On](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-js-sso) Microsoft docs page for the different scenarios in which MSAL.js enables single sign-on.
+
+## What browsers is MSAL.js supported on?
+
+MSAL.js has been tested with the following browsers:
+
+IE 11, Edge, Chrome, Firefox and Safari
+
+Keep [these steps](./docs/msal-in-IE.md) in mind when using MSAL.js with IE.
+There are certain known issues and mitigations documented for Safari, IE and Edge. Please check out:
+* [Known issues on IE and Edge](./docs/msal-in-IE-and-edge.md#issues-due-to-security-zones)
+* [Known issue on Safari](./docs/safari-issues.md)
+
+## How to avoid page reloads when acquiring and renewing tokens silently?
+
+MSAL.js uses hidden iframes to acquire and renew tokens silently in the background. Azure AD returns the token back to the registered `redirect_uri` specified in the token request(by default this is the app's root page). Since the response is a 302, it results in the HTML corresponding to the `redirect_uri` getting loaded in the iframe. Usually the app's redirect_uri is the root page and this causes it to reload.
+
+In other cases, if navigating to the app's root page requires authentication, it might lead to **nested iframes** or **xframe deny error**.
+
+Since, MSAL.js cannot dismiss the 302 issued by Azure AD and is required to process the returned token, it cannot prevent the `redirect_uri` from getting loaded in the iframe.
+
+To avoid the entire app reloading again or other errors caused due to this, please follow these workarounds:
+
+* **Specify a different html for the iframe:**
+
+    Set `redirect_uri` property on config to a simple page, that does not require authentication. You have to make sure that it matches with the `redirect_uri` registered in Azure portal. This will not affect user's login experience as MSAL saves the start page when user begins the login process and redirects back to the exact location after login is completed.
```

Comment:
This needs to be more specific (explicitly mention that it requires `msal@1.2.0` and include a simple example.